### ISSUE TITLE
Decoding DNS answer rrdata for MX records

### ISSUE THREAD

**Issue by RYNEQ** [secdev/scapy#2127]
I'm using scapy's sniffer to get DNS packets and it works like a charm  
The problem is when I receive MX record `rrdata` are coded (struct.pack or something like that)  
How can I decrypt this data ?

For example I sniffed DNS for google MX request:
```python
def pkt_callback(pkt):
    ip = pkt['IP']
    dns = pkt['DNS']
    for i in range(dns.ancount):
            dnsrr = dns.an[i]
            print(dnsrr.rdata)

sniff(iface='enp0s25', prn=pkt_callback, filter="udp and ( port 53 )", store=0)
```
I get this:
```python
b'\x00\n\x05aspmx\x01l\xc0\x0c'
b'\x002\x04alt4\xc0*'
b'\x00\x1e\x04alt2\xc0*'
b'\x00\x14\x04alt1\xc0*'
b'\x00(\x04alt3\xc0*'
```
Which should be this:
```python
10 aspmx.l.google.com.
50 alt4.aspmx.l.google.com.
30 alt2.aspmx.l.google.com.
20 alt1.aspmx.l.google.com.
40 alt3.aspmx.l.google.com.
```
I was able to extract "**perf**" field with this:
```python
perf = struct.unpack('!h', rrdata[0:2])[0]
```

But the rest of data is not that eas; my best afford was this:
```python
def decode_mx_rdata(rdata):
    perf = struct.unpack('!h', rdata[0:2])[0]
    rdata = rdata[2:]

    parts = []
    while rdata:
        if len(rdata) % 2 == 0:
            l = rdata[0:2]
            p = 2
        else:
            l = rdata[0:1]+b'\x00'
            p = 1
        l = int(struct.unpack('h', l)[0])
        parts.append(rdata[p:l+p])
        rdata = rdata[l+p:]
    return perf, parts
```
How can I correctly decode this data?

**Comment by gpotter2** [comment]
Thanks for the report !

Could you share a full packet (pcap or hex or bytes) ? It'll be much easier

**Comment by gpotter2** [comment]
Reproducible with
```python
sr1(IP(dst="8.8.8.8")/UDP(sport=RandShort(), dport=53)/DNS(rd=1,qd=DNSQR(qname="google.com",qtype="MX")))
```

**Comment by gpotter2** [comment]
@RYNEQ Fixed in https://github.com/secdev/scapy/pull/2128. please have a look

**Comment by RYNEQ** [comment]
> @RYNEQ Fixed in #2128. please have a look

I confirm that this fixes the bug; Thanks
